PostHogTests/TestUtils/MockPostHogServer.swift

```diff
@@ -14,73 +14,104 @@ import OHHTTPStubsSwift
+                "disabled-flag": false,
+                "number-value": true,
+            ]
+
+            if self.errorsWhileComputingFlags {
+                flags["new-flag"] = true
+                flags.removeValue(forKey: "bool-value")
+            }
+
             let obj: [String: Any] = [
-                "featureFlags": [
-                    "bool-value": true,
-                    "string-value": "test",
-                ],
+                "featureFlags": flags,
                 "featureFlagPayloads": [
                     "payload-bool": "true",
-                    "payload-number": "2",
+                    "number-value": "2",
                     "payload-string": "\"string-value\"",
                     "payload-json": "{ \"foo\": \"bar\" }",
                 ],
+                "errorsWhileComputingFlags": self.errorsWhileComputingFlags,
             ]
 
             return HTTPStubsResponse(jsonObject: obj, statusCode: 200, headers: nil)
         }
 
         stub(condition: isPath("/batch")) { _ in
-            HTTPStubsResponse(jsonObject: ["status": "ok"], statusCode: 200, headers: nil)
+            if self.return500 {
+                HTTPStubsResponse(jsonObject: [], statusCode: 500, headers: nil)
+            } else {
+                HTTPStubsResponse(jsonObject: ["status": "ok"], statusCode: 200, headers: nil)
+            }
         }
 
         HTTPStubs.onStubActivation { request, _, _ in
             if request.url?.path == "/batch" {
-                self.trackRequest(request)
+                self.trackBatchRequest(request)
+            } else if request.url?.path == "/decide" {
+                self.trackDecide(request)
             }
         }
     }
 
-    func start() {
+    func start(batchCount: Int = 1) {
+        reset(batchCount: batchCount)
+
         HTTPStubs.setEnabled(true)
     }
 
     func stop() {
-        requests = []
-        HTTPStubs.removeAllStubs()
-    }
+        reset()
 
-    func expectation(_ requestCount: Int) -> XCTestExpectation {
-        expectation = XCTestExpectation(description: "\(requestCount) requests to occur")
-        expectationCount = requestCount
-
-        return expectation!
+        HTTPStubs.removeAllStubs()
     }
 
-    var posthogConfig: PostHogConfig {
-        let config = PostHogConfig(apiKey: "test-123", host: "http://localhost:9001")
-
-        return config
+    func reset(batchCount: Int = 1) {
+        batchRequests = []
+        decideRequests = []
+        batchExpectation = XCTestExpectation(description: "\(batchCount) batch requests to occur")
+        decideExpectation = XCTestExpectation(description: "1 decide requests to occur")
+        batchExpectationCount = batchCount
+        errorsWhileComputingFlags = false
+        return500 = false
     }
 
-    func parseBatchRequest(_ context: URLRequest) -> [String: Any]? {
+    func parseRequest(_ context: URLRequest, gzip: Bool = true) -> [String: Any]? {
```

Comment:
I wonder if decide can be `gzip` as well? but likely not important since its just a small payload.

Reply:
Nope decide can't be gzipped, only ingestion currently (although it sounds like we might move away from that)